course_catalog/etl/mitx.py

```diff
@@ -25,6 +34,72 @@ def _is_mit_course(course):
     return False
 
 
+def get_most_relevant_run(runs):
+    """
+    Helper function to determine the most relavant course run.
+
+    Args:
+        runs (QuerySet): a set of LearningResourseRun objects
+    Returns:
+        A LearningResourseRun object
+    """
+
+    # if there is a current run in the set pick it
+    most_relavant_run = next(
+        (run for run in runs if run.availability == AvailabilityType.current.value),
+        None,
+    )
+
+    if not most_relavant_run:
+        # if there a future runs in the set, pick the one with earliest start date
+        runs = runs.order_by("best_start_date")
+        most_relavant_run = next(
+            (
+                run
+                for run in runs
+                if run.availability
+                in [
+                    AvailabilityType.upcoming.value,
+                    AvailabilityType.starting_soon.value,
+                ]
+            ),
+            None,
+        )
+
+        if not most_relavant_run:
+            # get latest past run by start date
+            most_relavant_run = next((run for run in runs.reverse()))
+
+    return most_relavant_run
+
+
+def dedupe_courses():
```

Comment:
Is this MITx-specific logic or can we generalize it to other platforms? I'm specifically thinking about how we occasionally archive MITx courses off to OLL, so we could start seeing them show up there as well.